base/sysinfo.jl

```diff
@@ -532,7 +533,26 @@ including e.g. a WebAssembly JavaScript embedding in a web browser.
 """
 isjsvm(os::Symbol) = (os === :Emscripten)
 
-for f in (:isunix, :islinux, :isbsd, :isapple, :iswindows, :isfreebsd, :isopenbsd, :isnetbsd, :isdragonfly, :isjsvm)
+"""
+    Sys.detectwsl([os])
+
+Runtime predicate for testing if Julia is running inside
+Windows Subsystem for Linux (WSL).
+
+!!! note
+    Unlike `Sys.iswindows`, `Sys.islinux` etc., this is a runtime tests, and thus
+    cannot meaningfully be used in `@static if ` constructs.
+
+!!! compat "Julia 1.12"
+    This function requires at least Julia 1.12.
+"""
+function detectwsl(os::Symbol)
+    islinux(os) &&
```

Comment:
Since this function can only determine whether the current system is WSL, it can't correctly query based on an arbitrary kernel, I don't think it makes sense to include the `os` argument.
```suggestion
function detectwsl()
    islinux() &&
```